app.js

```diff
@@ -64,6 +65,27 @@ function showCartHandler() {
   });
 }
 
+function updateCartTotal() {
+  let cartItems = document.querySelectorAll(".cart-item");
+  let cartTotal = document.querySelector(".cart-total");
+  let itemTotal;
+  let allItemsTotalArray = [];
+
+  for (let i = 0; i < cartItems.length; i++) {
+    const itemPrice = cartItems[i].querySelector("h5").innerHTML.slice(1);
+    const itemQuantity = cartItems[i].querySelector(".item-amount").innerHTML;
+    itemTotal = itemPrice * itemQuantity;
+
+    allItemsTotalArray.push(itemTotal);
+    const initialValue = 0;
+    const allItemsTotal = allItemsTotalArray.reduce(
```

Comment:
this logic of showing the total to the cart will run in every loop, right? What if we extract this logic of summing up all the item totals outside of this loop, will that be more efficient b/c it will run just once?

Reply:
i tried this too, but the total of the cart remained 0 every time outside the loop.

Reply:
let finalTotal = 0;
for(...) {
  // some logic to calculate total goes here
  finalTotal += total;
}
log(finalTotal);


Why this would be 0 if we access it within the same scope?

Reply:
I don't know why , but it did not work this way. we should discuss it together.

Reply:
ok. meanwhile try to debug it... put logs or breakpoints to understand why it's happening, should be a logical error from you.